src/test_suite/multiprocessing_utils.py

```diff
@@ -50,15 +53,15 @@ def process_target(
 
     # Process the output
     output_data = bytearray(globals.output_buffer_pointer[: out_sz.value])
-    output_object = globals.harness_ctx.effects_type()
+    output_object = harness_ctx.effects_type()
     output_object.ParseFromString(output_data)
 
     return output_object
 
 
-def read_context_serialized(test_file: Path) -> str | None:
+def read_context_serialized(harness_ctx: HarnessCtx, test_file: Path) -> str | None:
     """
-    Reads in test files and generates an InstrContext Protobuf object for a test case.
+    Reads in test files and generates an Context Protobuf object for a test case.
```

Comment:
nit/rant: it is bizarre that we need this function to read in a file, deserialize it, and then re-serialize it again just because of the human-readable encoding problem. We should just be calling `test_file.read_text()`

Reply:
yah we should change this in the next clean up PR